Domain/Problem/src/main/java/tw/waterball/judgegirl/problem/domain/repositories/ProblemQueryParams.java

```diff
@@ -22,26 +22,25 @@
  * @author - johnny850807@gmail.com (Waterball)
  */
 public class ProblemQueryParams {
-    public static final ProblemQueryParams NO_PARAMS = new ProblemQueryParams(null, null, false);
+    public static final ProblemQueryParams NO_PARAMS = new ProblemQueryParams(null, null, false, true, true);
 
     private final String[] tags;
 
     @Nullable
     private final Integer page;
 
-    private final boolean excludeArchive;
+    private final boolean archive;
 
-    private final boolean includeInvisibleProblems;
+    private final boolean visible;
 
-    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean includeInvisibleProblems) {
-        this(tags, page, true, includeInvisibleProblems);
-    }
+    private final boolean invisible;
 
-    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean excludeArchive, boolean includeInvisibleProblems) {
+    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean archive, boolean visible, boolean invisible) {
```

Comment:
rename `archive` to `queryArchive`
rename `visible` to `queryVisible`
rename `invisible` to `queryInvisible`

and add a `javadoc` to explain the parameters